lib/features/admins/presentation/pages/admins_screen.dart

```diff
@@ -0,0 +1,102 @@
+import 'package:auto_route/auto_route.dart';
+import 'package:fiura_ecosystem/dependencies.dart';
+import 'package:fiura_ecosystem/features/admins/presentation/cubit/admin_cubit.dart';
+import 'package:fiura_ecosystem/features/admins/presentation/cubit/admin_state.dart';
+import 'package:fiura_ecosystem/features/widgets/empty_list_widget.dart';
+import 'package:fiura_ecosystem/features/widgets/on_load_message.dart';
+import 'package:fiura_ecosystem/features/admins/presentation/widgets/remove_admin_modal.dart';
+import 'package:fiura_ecosystem/features/widgets/tile_image_widget.dart';
+import 'package:fiura_ecosystem/router/app_router.gr.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+
+class AdminScreen extends StatelessWidget {
+  const AdminScreen({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return BlocProvider(
+      create: (_) => getIt<AdminCubit>()..getAdmins(),
+      child: BlocBuilder<AdminCubit, AdminState>(
+        builder: (context, state) {
+          return state.maybeWhen(
+            loading: () => const Center(child: OnLoadMessage()),
+            error: (error) => Center(
+              child: Center(child: Text(error)),
+            ),
+            loadData: (adminList) => Scaffold(
+              floatingActionButton: FloatingActionButton(
+                onPressed: () {
+                  context.router.push(const CreateAdminScreenRoute());
+                },
+                child: const Icon(Icons.add),
+              ),
+              body: adminList.isNotEmpty
+                  ? FutureBuilder<void>(
+                      future: Future.wait(
+                        adminList.map(
+                          (admin) =>
+                              precacheImage(NetworkImage(admin.photo), context),
+                        ),
+                      ),
```

Comment:
I have been thinking about this implementation and my issue is that iof the page has 1000 photos the view has to wait until precache all the 1000 to show the view. so could be slow in certain cases and is also possible that you are loading 90% stuff that the user will no see

Reply:
fixed